### PR TITLE
Interrupt property

### LINKED ISSUE #5: Typing mismatch
When using both comsync and pyodide-worker-runner, it is not possible to assign a value of PyodideClient to a variable of type SyncClient with the following error:
`
Type 'PyodideClient<any>' is not assignable to type 'SyncClient<any>'.   Types have separate declarations of a private property '_interruptRejector'.
`
I believe this is caused due to a dependency mismatch, as pyodide-worker-runner uses a currently outdated version of comsync (0.0.6 instead of 0.0.7). Updating this should be enough. If not, the private fields should be made protected, but I doubt this will be required.

### PR DESCRIPTION
**Pull request by winniederidder**
Turns comsync into a peer dependency
Implements the [discussed functionality](https://github.com/alexmojaki/futurecoder/issues/328)
Closes #5

### CODE REVIEW

**Comment by winniederidder** [comment]
I realised a bit late I am still on my fork, but this shouldn't pose a real issue.

**Comment by winniederidder** [comment]
@alexmojaki Any idea why the Browserstack setup fails?

File: lib/index.ts [hunk 1]
```diff
@@ -114,22 +114,21 @@ export function makeRunnerCallback(
...
   ): Promise<R> {
-    const pyodide = await Promise.resolve(pyodidePromise);
-
-    if (interruptBuffer) {
-      pyodide.setInterruptBuffer(interruptBuffer);
-    }
-
-    return func(comsyncExtras, ...args);
+    const pyodideExtras: PyodideExtras = { ...comsyncExtras };
+    pyodideExtras.interruptBuffer = interruptBuffer;
```
Comment (alexmojaki) [inline lib/index.ts:130]:
```suggestion
    const pyodideExtras: PyodideExtras = { ...comsyncExtras, interruptBuffer };
```

File: lib/index.ts [hunk 2]
```diff
@@ -114,22 +114,21 @@ export function makeRunnerCallback(
   };
 }
 
+export interface PyodideExtras extends SyncExtras {
+  interruptBuffer?: Int32Array;
```
Comment (alexmojaki) [inline lib/index.ts:118]:
```suggestion
  interruptBuffer: Int32Array | null;
```

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -35,7 +35,9 @@ jobs:
           path: ${{ steps.yarn-cache.outputs.dir }}
           key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
       - name: Yarn install
-        run: yarn --frozen-lockfile
+        run: |
+          yarn add comsync
```
Comment (alexmojaki) [inline .github/workflows/CI.yml:39]: Add it to dev dependencies

**Review by alexmojaki — CHANGES_REQUESTED** [review]
Forks don't have access to secrets which are needed for Browserstack, that's why PRs need to be on a branch.

Also `yarn run format` so that whitespace etc. remains consistent.

**Pull request closed without merging**